server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIT.java

```diff
@@ -505,6 +507,124 @@ public void testNodeRolesWithDataNodeLegacySettings() throws ExecutionException,
         assertEquals(expectedNodesRoles, Set.of(getNodeRoles(client, 0), getNodeRoles(client, 1)));
     }
 
+    public void testClusterStatsMetricFiltering() {
+        internalCluster().startNode();
+        ensureGreen();
+
+        client().admin().indices().prepareCreate("test1").setMapping("{\"properties\":{\"foo\":{\"type\": \"keyword\"}}}").get();
+
+        ClusterStatsResponse response = client().admin()
+            .cluster()
+            .prepareClusterStats()
+            .useAggregatedNodeLevelResponses(randomBoolean())
+            .requestMetrics(Set.of(Metric.FS.metricName(), Metric.JVM.metricName(), Metric.PLUGINS.metricName(), Metric.OS.metricName()))
+            .applyMetricFiltering(true)
+            .get();
+        assertNotNull(response.getNodesStats());
+        assertNotNull(response.getNodesStats().getJvm());
+        assertNotNull(response.getNodesStats().getOs());
+        assertNotNull(response.getNodesStats().getPlugins());
+        assertNotNull(response.getNodesStats().getFs());
+        assertNull(response.getIndicesStats());
+
+        response = client().admin()
+            .cluster()
+            .prepareClusterStats()
+            .useAggregatedNodeLevelResponses(randomBoolean())
+            .requestMetrics(Set.of(Metric.INDICES.metricName()))
+            .indexMetrics(IndexMetrics.allIndicesMetrics())
+            .applyMetricFiltering(true)
+            .get();
+        assertNotNull(response.getIndicesStats());
+        assertNotNull(response.getIndicesStats().getMappings());
+        assertNotNull(response.getIndicesStats().getAnalysis());
+        assertNotNull(response.getIndicesStats().getCompletion());
+        assertNotNull(response.getIndicesStats().getQueryCache());
+        assertNotNull(response.getIndicesStats().getShards());
+        assertNotNull(response.getIndicesStats().getSegments());
+        assertNotNull(response.getIndicesStats().getStore());
+        assertNotNull(response.getIndicesStats().getFieldData());
+        assertNotNull(response.getIndicesStats().getIndexCount());
+        assertNull(response.getNodesStats());
+
+        response = client().admin().cluster().prepareClusterStats().useAggregatedNodeLevelResponses(randomBoolean()).get();
+        assertNotNull(response.getIndicesStats());
+        assertNotNull(response.getNodesStats());
+        assertNotNull(response.getIndicesStats().getMappings());
+        assertNotNull(response.getIndicesStats().getAnalysis());
+        assertNotNull(response.getNodesStats().getJvm());
+        assertNotNull(response.getNodesStats().getOs());
+
+        response = client().admin()
```

Comment:
Lets seperate it out as three tests and assert on the values also.

Reply:
Done.